toolchest_client/files/split.py

```diff
@@ -68,11 +72,49 @@ def split_file_by_lines(input_file_path, num_lines_in_group=4, max_bytes=4.5 * 1
             # Time to switch output files
             current_output_bytes = 0
             current_output_file.close()
-            yield current_output_file_path
+            yield current_line_number, current_output_file_path
             current_split_number += 1
             need_to_open_new_file = True
 
     current_output_file.close()
     large_input_file.close()
     if current_output_bytes != 0:
-        yield current_output_file_path
+        yield current_line_number, current_output_file_path
+
+
+def split_paired_files_by_lines(input_file_paths, num_lines_in_group=4, max_bytes=4.5 * 1024 * 1024 * 1024):
+    """Splits files by line, in groups of two (for paired end R1/R2 reads).
+    Defaults to splitting files by four line groups to support FASTA and FASTQ files.
+    Note that this is a generator.
+
+    :param input_file_paths: Path to the file which is to be split.
+    :param num_lines_in_group: Number of contiguous lines which cannot be split from one another.
+    :param max_bytes: Maximum size of each new file.
+    """
+
+    grouped_input_file_paths = {}
+    for input_file_path in input_file_paths:
+        no_read_file_path = re.sub(r'R\d', '', input_file_path)
+        if grouped_input_file_paths.get(no_read_file_path):
+            grouped_input_file_paths[no_read_file_path].append(input_file_path)
+        else:
+            grouped_input_file_paths[no_read_file_path] = [input_file_path]
+
+    for _, [read_one_file_path, read_two_file_path] in grouped_input_file_paths.items():
```

Comment:
If no pair is found (i.e., some reads are standalone single-end), is the resulting behavior intended?

Reply:
If no pair is found, we'd end up with something like:

```
[yields pairs until the non-matched file is found]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 2, got 1)
```

It's not ideal; ideally we should be confirming that there are an even number of matching R1s and R2s. Will resolve.